File: server-scala-play-server/app/api/ShipmentBatchApi.scala
package api

import play.api.libs.json._
import model.BigDecimal
import model.ShipmentBatch
import model.ShipmentImportStatus

@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
trait ShipmentBatchApi {
  /**
    * Create Shipment Batch
    * Create a new shipment batch
    */
  def createShipmentBatch(version: BigDecimal, body: Option[ShipmentBatch]): ShipmentBatch

  /**
    * Delete Shipment Batch
    * Search for shipment batches
    * @param batchId the id of the shipment batch to delete
    */
  def deleteShipmentBatch(version: BigDecimal, batchId: Long): Unit

  /**
    * Get Shipment Batch
    * Get an existing shipment batch
    * @param batchId the id of the shipment batch to get
    */
  def getShipmentBatch(version: BigDecimal, batchId: Long): ShipmentBatch

  /**
    * Get Shipment Batch Status
    * Get the import status list of the import shipment batch
    * @param batchId The id of the requested shipment batch
    * @param accountId the id of the logged in user
    * @param sortField The field to sort by
    * @param descending Determines whether the sorted list is in descending or ascending order
    * @param start The start index for pagination
    * @param limit The limit for pagination
    * @param valid Valid import status only or not
    * @param started Started import status only
    * @param completed Completed import status only
    * @param hasShipment Has shipment associate to the status
    * @param hasRoute Has route associate to the status
    * @param keyword The keyword to search for
    */
  def getShipmentBatchStatus(version: BigDecimal, batchId: Long, accountId: Long, sortField: String, descending: Boolean, start: Int, limit: Int, valid: Option[Boolean], started: Option[Boolean], completed: Option[Boolean], hasShipment: Option[Boolean], hasRoute: Option[Boolean], keyword: Option[String]): List[ShipmentImportStatus]

  /**
    * Search Shipment Batch
    * Search for shipment batches
    * @param hubId The associated service hub
    * @param sortField The field to sort by
    * @param descending Determines whether the sorted list is in descending or ascending order
    * @param start The start index for pagination
    * @param limit The limit for pagination
    */
  def searchShipmentBatch(version: BigDecimal, hubId: Long, sortField: String, descending: Boolean, start: Int, limit: Int): List[ShipmentBatch]
}

File: server-ruby-sinatra/api/shipment_batch_api.rb
require 'json'


MyApp.add_route('POST', '/api/3.18/shipment/batch', {
  "resourcePath" => "/ShipmentBatch",
  "summary" => "Create Shipment Batch",
  "nickname" => "create_shipment_batch",
  "responseClass" => "ShipmentBatch",
  "endpoint" => "/shipment/batch",
  "notes" => "Create a new shipment batch",
  "parameters" => [
    {
      "name" => "body",
      "description" => "",
      "dataType" => "ShipmentBatch",
      "paramType" => "body",
    }
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('DELETE', '/api/3.18/shipment/batch/{batchId}', {
  "resourcePath" => "/ShipmentBatch",
  "summary" => "Delete Shipment Batch",
  "nickname" => "delete_shipment_batch",
  "responseClass" => "void",
  "endpoint" => "/shipment/batch/{batchId}",
  "notes" => "Search for shipment batches",
  "parameters" => [
    {
      "name" => "batch_id",
      "description" => "the id of the shipment batch to delete",
      "dataType" => "Integer",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/shipment/batch/{batchId}', {
  "resourcePath" => "/ShipmentBatch",
  "summary" => "Get Shipment Batch",
  "nickname" => "get_shipment_batch",
  "responseClass" => "ShipmentBatch",
  "endpoint" => "/shipment/batch/{batchId}",
  "notes" => "Get an existing shipment batch",
  "parameters" => [
    {
      "name" => "batch_id",
      "description" => "the id of the shipment batch to get",
      "dataType" => "Integer",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/shipment/batch/{batchId}/status', {
  "resourcePath" => "/ShipmentBatch",
  "summary" => "Get Shipment Batch Status",
  "nickname" => "get_shipment_batch_status",
  "responseClass" => "Array<ShipmentImportStatus>",
  "endpoint" => "/shipment/batch/{batchId}/status",
  "notes" => "Get the import status list of the import shipment batch",
  "parameters" => [
    {
      "name" => "account_id",
      "description" => "the id of the logged in user",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "valid",
      "description" => "Valid import status only or not",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "started",
      "description" => "Started import status only",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "completed",
      "description" => "Completed import status only",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "has_shipment",
      "description" => "Has shipment associate to the status",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "has_route",
      "description" => "Has route associate to the status",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "keyword",
      "description" => "The keyword to search for",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "sort_field",
      "description" => "The field to sort by",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "descending",
      "description" => "Determines whether the sorted list is in descending or ascending order",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "The start index for pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "The limit for pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "batch_id",
      "description" => "The id of the requested shipment batch",
      "dataType" => "Integer",
      "paramType" => "path",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end


MyApp.add_route('GET', '/api/3.18/shipment/batch', {
  "resourcePath" => "/ShipmentBatch",
  "summary" => "Search Shipment Batch",
  "nickname" => "search_shipment_batch",
  "responseClass" => "Array<ShipmentBatch>",
  "endpoint" => "/shipment/batch",
  "notes" => "Search for shipment batches",
  "parameters" => [
    {
      "name" => "hub_id",
      "description" => "The associated service hub",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "sort_field",
      "description" => "The field to sort by",
      "dataType" => "String",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "descending",
      "description" => "Determines whether the sorted list is in descending or ascending order",
      "dataType" => "Boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "start",
      "description" => "The start index for pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "The limit for pagination",
      "dataType" => "Integer",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

